ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineGetPayloadV2.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.engine;
+
+import org.hyperledger.besu.consensus.merge.blockcreation.MergeMiningCoordinator;
+import org.hyperledger.besu.ethereum.ProtocolContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.results.BlockResultFactory;
+import org.hyperledger.besu.ethereum.core.Block;
+
+import io.vertx.core.Vertx;
+
+public class EngineGetPayloadV2 extends EngineGetPayload {
+
+  public EngineGetPayloadV2(
+      final Vertx vertx,
+      final ProtocolContext protocolContext,
+      final MergeMiningCoordinator mergeMiningCoordinator,
+      final BlockResultFactory blockResultFactory,
+      final EngineCallListener engineCallListener) {
+    super(vertx, protocolContext, mergeMiningCoordinator, blockResultFactory, engineCallListener);
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ENGINE_GET_PAYLOAD_V2.getMethodName();
+  }
+
+  @Override
+  protected JsonRpcResponse createResponse(final JsonRpcRequestContext request, final Block block) {
+    return new JsonRpcSuccessResponse(
+        request.getRequest().getId(), blockResultFactory.enginePayloadTransactionCompleteV2(block));
```

Comment:
I'm hoping we can make a backwards compatible V2 so that the only thing that needs to know about V2 is the very thinnest layer that maps RPC name to class.

Maybe we could pass in an `includeBlockValue` or `isShanghai` to trigger this behaviour in the shared layer below